crates/miden-lib/src/account/wallets/mod.rs

```diff
@@ -75,6 +78,37 @@ pub fn create_basic_wallet(
     Ok((account, account_seed))
 }
 
+// AUX WALLET
+// ================================================================================================
+
+/// An [`AccountComponent`] implementing an auxiliary wallet.
+///
+/// It reexports the procedures from `miden::contracts::wallets::aux`. When linking against this
+/// component, the `miden` library (i.e. [`MidenLib`](crate::MidenLib)) must be available to the
+/// assembler which is the case when using [`TransactionKernel::assembler()`][kasm]. This component
+/// contains a single procedure:
+/// - `add_asset_to_note`, which can be used to add an asset to the output note with the specified
+///   index. It allows for a so-called "in-flight" asset transfer, i.e., an asset that doesn't "pass
+///   through" the account, but is directly "moved" from one note to another.
+///
+/// All methods require authentication. Thus, this component must be combined with a component
+/// providing authentication.
+///
```

Comment:
```suggestion
```

This procedure does not require authentication because, authentication is only required when the account state changes and this doesn't change the state.

This might change with #1331.

Actually, this comment is also incorrect on `BasicWallet` since `create_note` does not require authentication - would you mind changing the docs there to indicate that?